modules/data.remote/R/soilgrids.R

```diff
@@ -0,0 +1,224 @@
+    
+    # create virtual rasters && extract SOC values - the original unit is dg/kg
+    soc_mean <- terra::extract(terra::rast(soc_mean.url), p_reproj)
+    soc_Q0.05_map <- terra::extract(terra::rast(soc_Q0.05.url), p_reproj)
+    soc_Q0.50_map <- terra::extract(terra::rast(soc_Q0.50.url), p_reproj)
+    soc_Q0.95_map <- terra::extract(terra::rast(soc_Q0.95.url), p_reproj)
+    # ---- could also replace using terra with just raster and extract similar to the landtrendr functions
+    
+    #change the unit to more common g/kg
+    soc_mean_real <- soc_mean[, 2] / 10
+    soc_Q0.05_real <- soc_Q0.05_map[, 2] / 10
+    soc_Q0.50_real <- soc_Q0.50_map[, 2] / 10
+    soc_Q0.95_real <- soc_Q0.95_map[, 2] / 10
+    
+    socquant[i,] <- c(soc_mean_real, soc_Q0.05_real, soc_Q0.50_real, soc_Q0.95_real)
+    siteids[i,] <- rep(as.numeric(internal_site_info$site_info.id), 4)
+    
+    rownames(socquant) <- c("0-5", "5-15", "15-30", "30-60", "60-100", "100-200")
+    colnames(socquant) <- c(rep("Mean", length(internal_site_info$site_info.lon)), 
+                            rep("0.05", length(internal_site_info$site_info.lon)),
+                            rep("0.5", length(internal_site_info$site_info.lon)),
+                            rep("0.95", length(internal_site_info$site_info.lon)))
+    rownames(siteids) <- c("0-5", "5-15", "15-30", "30-60", "60-100", "100-200")
+    colnames(siteids) <- c(rep("Mean", length(internal_site_info$site_info.lon)), 
+                           rep("0.05", length(internal_site_info$site_info.lon)), 
+                           rep("0.5", length(internal_site_info$site_info.lon)), 
+                           rep("0.95", length(internal_site_info$site_info.lon)))
+    i <- i + 1
+    
+    ### Display progress to console
+    if (verbose) {
+      setTxtProgressBar(pb, j)
+      j <- j+1
+      flush.console()
+    }
+    # cleanup interim results
+    rm(soc_mean.url, soc_Q0.05.url, soc_Q0.50.url, soc_Q0.95.url, 
+       soc_mean, soc_Q0.05_map, soc_Q0.50_map, soc_Q0.95_map,
+       soc_mean_real, soc_Q0.05_real, soc_Q0.50_real, soc_Q0.95_real)
+  }
+  
+  if (verbose) {
+    close(pb)
+  }
+  
+  # parse extracted data and prepare for output
+  soc_fit <- melt(socquant, id.vars = c("Mean"))
+  id_fit <- melt(siteids, id.vars = c("Mean"))
+  colnames(soc_fit) <- c("Depth", "Quantile", "Value")
+  soc_fit$Variable <- rep("soc", length(nrow(soc_fit)))
+  soc_fit$siteids <- id_fit$value
+  dat <- split(soc_fit, list(soc_fit$siteids, soc_fit$Depth))
+  
+  #assume the soc profile follows gamma distribution best
+  cgamma <- function(theta, val, stat) {
+    pred <- rep(NA, 4)
+    names(pred) = stat
+    if ("Mean" %in% stat) {
+      pred["Mean"] <- theta[1] / theta[2]
+    }
+    qstat <- as.numeric(stat)[!is.na(as.numeric(stat))]
+    pred[as.character(qstat)] <- qgamma(qstat, theta[1], theta[2])
+    return(sum((pred - val) ^ 2))
+  }
+  
+  fitQ <- function(x) {
```

Comment:
same

Reply:
as stated above I dont actually know if it makes sense to move these if these arent really used anywhere else?